truss-chains/truss_chains/stub.py

```diff
@@ -55,12 +59,19 @@ def predict_sync(self, json_payload):
                     logging.info(
                         f"Retrying `{self._service_descriptor.name}`, " f"attempt {num}"
                     )
-                return utils.handle_response(
-                    self._client_sync.post(
-                        self._service_descriptor.predict_url, json=json_payload
-                    ),
-                    self.name,
-                )
+                with tracer.start_as_current_span(name="predict_remote") as span:
```

Comment:
Maybe would be better for structuring to define a helper or custom context manager to set this up instead of inlining it here.